lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -3832,6 +3832,25 @@ void ConformanceChecker::checkNonFinalClassWitness(ValueDecl *requirement,
   }
 }
 
+// If the given witness matches a generic RawRepresentable function conforming
+// with a given protocol e.g. `func == <T : RawRepresentable>(lhs: T, rhs: T) ->
+// Bool where T.RawValue : Equatable`
+static bool isRawRepresentableGenericFunction(
+    ASTContext &ctx, const ValueDecl *witness,
+    const NormalProtocolConformance *conformance) {
+  auto *fnDecl = dyn_cast<AbstractFunctionDecl>(witness);
+  return fnDecl && fnDecl->isGeneric() &&
```

Comment:
Let's try matching more aggressively. We can ensure `isStdlibDecl` and that there aren't further requirements. This'll let users that have a global == witness of their own continue to compile against it.

Reply:
Make sense just fixed, thanks @CodaFi